File: Difficulty: Medium/Find nth root of m/find-nth-root-of-m.java
class Solution {
    
    public int func(int mid, int n, int m) {
        long ans = 1;
        
        for(int i=1; i<=n ; i++) {
            ans = ans * mid;
            
            if(ans > m) {
                return 2;
            }
        }
        if(ans == m) return 1;
        return 0;
    }
    
    public int nthRoot(int n, int m) {
        // code here
        if(m==0 )return 0;
        
        int low = 1;
        int high = m;
        
        while(low<=high) {
            int mid = low + (high - low) / 2;
            int midnum = func(mid, n, m);
            
            if(midnum == 1 ) {
                return mid;
            }
            else if (midnum == 0) low = mid+1;
            else high = mid-1;
            
        }
        
        
        return -1;    
    }
}

File: Difficulty: Medium/Find nth root of m/find-nth-root-of-m.cpp
class Solution {
  public:
  
    int func(int mid, int n, int m) {
        long long answer = 1;
        
        for(int i=1; i<=n; i++) {
            answer = answer * mid;
            
            if(answer > m) return 2;
        }
        if(answer == m) return 1;
        return 0;
    }
  
    int nthRoot(int nth, int num) {
        // Code here
        if(num == 0) return 0;
        
        int low = 1;
        int high = num;
        
        while(low<= high ) {
            int mid = low + (high - low) / 2;
            
            int midnum = func(mid, nth, num) ;
            
            if(midnum == 1) {
                return mid;
            }
            
            else if (midnum == 0) low = mid + 1;
            else high = mid-1;
        }
        return -1;
    }
};